src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  // static alive = [];
+
+  constructor(name, health = 100,) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    if (this.hidden === false) {
+      this.hidden = true;
+    } else if (this.hidden === true) {
+      this.hidden = false;
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herb) {
+    if (herb instanceof Herbivore && herb.hidden === false) {
+      herb.health -= 50;
+    }
+
+    if (herb.health === 0) {
+      Animal.alive.splice(Animal.alive.health);
```

Comment:
Let's use the `filter` method instead of splice and get rid of the redundant `if` check.
Just filter your animals whos health > 0